nixos/modules/services/networking/rosenpass.nix

```
@@ -0,0 +1,220 @@
+        freeformType = settingsFormat.type;
+
+        options = {
+          public_key = mkOption {
+            type = path;
+            description = "Path to a file containing the public key of the local Rosenpass peer. Generate this by running {command}`rosenpass gen-keys`.";
+          };
+
+          secret_key = mkOption {
+            type = path;
+            description = "Path to a file containing the secret key of the local Rosenpass peer. Generate this by running {command}`rosenpass gen-keys`.";
+          };
+
+          listen = mkOption {
+            type = listOf str;
+            description = "List of local endpoints to listen for connections.";
+            default = [ ];
+            example = literalExpression "[ \"0.0.0.0:10000\" ]";
+          };
+
+          verbosity = mkOption {
+            type = enum [ "Verbose" "Quiet" ];
+            default = "Quiet";
+            description = "Verbosity of output produced by the service.";
+          };
+
+          peers =
+            let
+              peer = submodule {
+                freeformType = settingsFormat.type;
+
+                options = {
+                  public_key = mkOption {
+                    type = path;
+                    description = "Path to a file containing the public key of the remote Rosenpass peer.";
+                  };
+
+                  endpoint = mkOption {
+                    type = nullOr str;
+                    default = null;
+                    description = "Endpoint of the remote Rosenpass peer.";
+                  };
+
+                  device = mkOption {
+                    type = str;
+                    default = cfg.defaultDevice;
+                    defaultText = literalExpression "config.${opt.defaultDevice}";
+                    description = "Name of the local WireGuard interface to use for this peer.";
+                  };
+
+                  peer = mkOption {
+                    type = str;
+                    description = "WireGuard public key corresponding to the remote Rosenpass peer.";
+                  };
+
+                  extra_params = mkOption {
+                    type = listOf str;
+                    default = [ ];
+                    description = "Extra WireGuard parameters to set. In NixOS, you would probably not want to set this and configure WireGuard via {option}`systemd.network.netdevs`.";
+                  };
+                };
+              };
+            in
+            mkOption {
+              type = listOf peer;
+              description = "List of peers to exchange keys with.";
+              default = [ ];
+            };
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for Rosenpass, see
+        <link xlink:href="https://rosenpass.eu/"/>
+        for further information.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable (
+    let
+      settings = cfg.settings;
+    in
+    {
```

```suggestion
  config = mkIf cfg.enable ({
```
having not to many aliases keeps the code readable and frankly is so short that we can write it out or make it more local.